src/components/RecommendationPage.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import axiosInstance from '../apis/axiosClient';
+import { Typography, CircularProgress, List, ListItem } from '@mui/material';
+import { useAuth } from '@akosasante/react-auth-context';
+const RecommendationPage = () => {
+  const { user } = useAuth();
+  const userId = user ? user.userId : null;
+
+  const [recommendations, setRecommendations] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (userId) {
+      axiosInstance
+        .get(`/recommendations/${userId}`)
+        .then((response) => {
+          setRecommendations(response.data.recommendations);
+          setLoading(false);
+        })
+        .catch((error) => {
+          console.error('Error fetching recommendations:', error);
+          setError('An error occurred while fetching recommendations.');
+          setLoading(false);
+        });
+    } else {
+      setLoading(false);
+    }
+  }, [userId]);
+
+  return (
```

Comment:
It'd be nice if we show some text, if we have fetched the recommendations, but the result is an empty array. Something to tell the customer "Sorry we don't have any recommendations for you yet, as you add more albums to your library we will be able to generate some" or something.

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/76d5ac49-f346-4958-ac1b-7f023e0e38ce)

Reply:
Looks good!